### ISSUE TITLE
Timeout on task creation

### ISSUE THREAD

**Issue by thundersen** [influxdata/influxdb-client-csharp#261]
__Steps to reproduce:__

I can successfully create a task using `curl` like this:
```
curl -v  --request POST 'localhost:8086/api/v2/tasks' \
  --header 'Content-Type: application/json' \
  --header 'Authorization: Token MyToken' \
  --data-raw '{
    "flux": "import \"experimental\"\nimport \"influxdata/influxdb/monitor\"\nimport \"influxdata/influxdb/v1\"\n\noption task = {\n    name: \"manual api test\",\n    every: 1m,\n}\n\nsettlement = \"Brummsack\"\ncheck_duration = 2d\n\nlastTime = (tables=<-) => {\n    extract = tables\n        |> last()\n        |> findColumn(fn: (key) => true, column: \"_time\")\n\n    return extract[0]\n}\n\nstop =
from(bucket: \"vevantzedata\")\n    |> range(start: 2100-01-01T00:00:00Z, stop: 2200-01-01T00:00:00Z)\n    |> filter(fn: (r) => r[\"_measurement\"] == \"stocks\")\n    |> filter(fn: (r) => r[\"settlement\"] == settlement)\n    |> filter(fn: (r) => r[\"_field\"] == \"Berries\")\n    |> lastTime()\n\nstart = experimental.subDuration(d: check_duration, from: stop)\n\nberries = from(bucket: \"vevantzedata\")\n    |> range(start: start, stop: stop)\n    |> filter(fn: (r) => r[\"_measurement\"] == \"stocks\")\n    |> filter(fn: (r) => r[\"settlement\"] == settlement)\n    |> filter(fn: (r) => r[\"_field\"] == \"Berries\")\n    |> aggregateWindow(every: check_duration, fn: mean, createEmpty: false)\n\nberries\n    |> v1[\"fieldsAsCols\"]()\n    |> monitor.check(\n        data: {\n            _check_id: \"0000000000000001\",\n            _check_name: task.name,\n            _type: \"threshold\",\n
     tags: {\"settlement\": settlement},\n        },\n        messageFn: (r) => \"Check: ${r._check_name} is: ${r._level} (${r.Berries}) for ${r.district} in ${r.settlement}\",\n        crit: (r) => r[\"Berries\"] <= 20.0,\n        warn: (r) => r[\"Berries\"] <= 100.0,\n        ok: (r) => r[\"Berries\"] > 100.0,\n    )",
    "org": "thundersen",
    "status": "active",
    "description": "manual api test"
}'
```
curl returns with `HTTP/1.1 201 Created` almost instantly.

I'm trying to create the same task using the C# library like this:
```
var options = new InfluxDBClientOptions.Builder()
                .Org(_org)
                .Bucket(_bucket)
                .Url(config.InfluxDBAddress)
                .AuthenticateToken(config.InfluxDBToken)
                .LogLevel(config.InfluxDBClientLogLevel)
                .TimeOut(TimeSpan.FromSeconds(60))
                .ReadWriteTimeOut(TimeSpan.FromSeconds(60))
                .Build();

var client = InfluxDBClientFactory.Create(options);

var tasks = _client.GetTasksApi();

var task = new TaskCreateRequest(
                    org: _org, 
                    flux: "same query as in curl", 
                    description: "Test Task");

tasks.CreateTaskAsync(task).GetAwaiter().GetResult();
```

__Expected behavior:__
Task gets created and the C# program continues.

__Actual behavior:__
The operation times out and I find this in the log:
```
[Info   :     Trace] --> POST /api/v2/tasks
[Info   :     Trace] --> Header: Accept=application/json
[Info   :     Trace] --> Header: Authorization=Token MyToken
[Info   :     Trace] --> Body: {"org":"thundersen","flux":"import \"experimental\"\nimport \"influxdata/influxdb/monitor\"\nimport \"influxdata/influxdb/v1\"\n\noption task = {\n    name: \"Berries Test Alert for Brummsack\",\n    every: 1m,\n}\n\nsettlement = \"Brummsack\"\ncheck_duration = 2d\n\nlastTime = (tables=<-) => {\n    extract = tables\n        |> last()\n        |> findColumn(fn: (key) => true, column: \"_time\")\n\n    return extract[0]\n}\n\nstop = from(bucket: \"vevantzedata\")\n    |> range(start: 2100-01-01T00:00:00Z, stop: 2200-01-01T00:00:00Z)\n    |> filter(fn: (r) => r[\"_measurement\"] == \"stocks\")\n    |> filter(fn: (r) => r[\"settlement\"] == settlement)\n    |> filter(fn: (r) => r[\"_field\"] == \"Berries\")\n    |> lastTime()\n\nstart = experimental.subDuration(d: check_duration, from: stop)\n\nberries = from(bucket: \"vevantzedata\")\n    |> range(start: start, stop: stop)\n    |> filter(fn: (r) => r[\"_measurement\"] == \"stocks\")\n    |> filter(fn: (r) => r[\"settlement\"] == settlement)\n    |> filter(fn: (r) => r[\"_field\"] == \"Berries\")\n    |> aggregateWindow(every: check_duration, fn: mean, createEmpty: false)\n    |> last()","description":"Test Task"}
[Info   :     Trace] --> END
[Info   :     Trace] -->
[Info   :     Trace] <-- 0
[Info   :     Trace] <-- END
[Error  : Unity Log] WebException: The operation has timed out.
Stack trace:
System.Net.HttpWebRequest.RunWithTimeoutWorker[T] (System.Threading.Tasks.Task`1[TResult] workerTask, System.Int32 timeout, System.Action abort, System.Func`1[TResult] aborted, System.Threading.CancellationTokenSource cts) (at <bcd022427bf24cb2b0d71a065670a2a4>:0)
System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) (at <bcd022427bf24cb2b0d71a065670a2a4>:0)
RestSharp.Http.GetRawResponseAsync (System.IAsyncResult result, System.Action`1[T] callback) (at <483801266a064cf6a016c4137514860b>:0)
RestSharp.Http.ResponseCallback (System.IAsyncResult result, System.Action`1[T] callback) (at <483801266a064cf6a016c4137514860b>:0)
Rethrow as HttpException: The operation has timed out.
InfluxDB.Client.Api.Service.TasksService.PostTasksAsyncWithIRestResponse (InfluxDB.Client.Api.Domain.TaskCreateRequest taskCreateRequest, System.String zapTraceSpan, System.Threading.CancellationToken cancellationToken) (at <7a0b5160427042679fafbf9769dfa37a>:0)
InfluxDB.Client.Api.Service.TasksService.PostTasksAsyncWithHttpInfo (InfluxDB.Client.Api.Domain.TaskCreateRequest taskCreateRequest, System.String zapTraceSpan, System.Threading.CancellationToken cancellationToken) (at <7a0b5160427042679fafbf9769dfa37a>:0)
InfluxDB.Client.Api.Service.TasksService.PostTasksAsync (InfluxDB.Client.Api.Domain.TaskCreateRequest taskCreateRequest, System.String zapTraceSpan, System.Threading.CancellationToken cancellationToken) (at <7a0b5160427042679fafbf9769dfa37a>:0)
Rethrow as AggregateException: One or more errors occurred. (The operation has timed out.)
System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) (at <3dd53bbd0ae9463fb53361469bbd0679>:0)
System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) (at <3dd53bbd0ae9463fb53361469bbd0679>:0)
System.Threading.Tasks.Task`1[TResult].get_Result () (at <3dd53bbd0ae9463fb53361469bbd0679>:0)
VeVantZeData.Collector.Alerting.InfluxDBAlerter+<>c__DisplayClass5_0.<Init>b__0 (System.Threading.Tasks.Task`1[TResult] t) (at <49afd60e294240ac865f0380613e5ed9>:0)
```

__Specifications:__
 - Client Version: 3.0.0
 - InfluxDB Version: 2.0.9
 - Platform: CLR 4.0.30319.42000 running Unity v2022.1.0.2267647 under Proton 6.3-7 on Ubuntu 20.4 - Linux 5.4.0-88-generic (I know it's a weird setup, but hey, writing data to InfluxDB has worked just fine so far! :)

**Comment by bednar** [comment]
Hi @thundersen,

thanks for using our client.

I've try your code and everything works correctly:

```csharp
Client.SetLogLevel(LogLevel.Body);

const string query = "from(bucket: \"my-bucket\")\n\t|> range(start: 0)\n\t|> last()";
var flux = $"option task = {{\nname: \"task-name\",\nevery: 1h\n}}\n\n{query}";

var request = new TaskCreateRequest(org: _organization.Name, description: "testing task", flux: flux);

var task = _tasksApi.CreateTaskAsync(request).GetAwaiter().GetResult();
Console.WriteLine($"{task}");
```

Is something interesting in InfluxDB logs?

Regards

**Comment by thundersen** [comment]
Hey @bednar, 

thanks a lot for your fast reply! 

Unfortunately I can't see anything useful in the logs.

That said, I'm very new to InfluxDB and might be missing something. As far as I know Influx running in Docker will log everything to StdOut by default, so that's what I'm looking at using `docker logs`.
I was hoping for an http access log, but couldn't find one.

I will go and see, if I can change InfluxDB's log level somewhere. Maybe that will yield something.

**Comment by thundersen** [comment]
Ok, so I changed the log level to `debug` and now I see this log line:
`ts=2021-11-05T10:38:41.169861Z lvl=debug msg=Request log_id=0XcumvPl000 service=http method=POST host=localhost:8086 path=/api/v2/tasks query= proto=HTTP/1.1 status_code=201 response_size=2480 content_length=1780 referrer= remote=192.168.48.1:32848 user_agent=influxdb-client-csharp took=36.529ms body="{\"org\":\"thundersen\",\"flux\":\"[MY QUERY]\",\"description\":\"Test Task\"}"`

The task is actually created and running successfully. Sorry, I missed that earlier. But the client still times out and my program crashes.

**Comment by thundersen** [comment]
By the way, I can't find the task description I'm assigning anywhere. Neither in the UI, the task execution logs, nor when listing tasks using the CLI (`influx task list`). Is that actually stored anywhere?

**Comment by thundersen** [comment]
@bednar, I tried it myself with a minimal .Net 5.0 console app using the same code. That works. So it's got to have something to do with my runtime environment.
If you have any idea what could be the cause for this (e.g. http settings I could try), please do comment. Otherwise feel free to just close this.

**Comment by bednar** [comment]
Hi @thundersen,

Did you find the source of your problem? It can be useful for other users...

Regards

**Comment by thundersen** [comment]
hi @bednar,

Unfortunately I didn't. Since this wasn't my only problem I decided to not use the library directly and instead wrap it in an API running in its own Docker container.
(The other problem was that the game that I'm developing a mod for ships with an ancient version of `Newtonsoft.Json` and I couldn't find a good way to redirect the binding to a newer version at runtime. I worked around that by compiling my own version of the client where I manually replaced the call to `JObject.ContainsKey` [here](https://github.com/influxdata/influxdb-client-csharp/blob/92ccdced2e77b145704c927dd3166c6e46b78d9f/Client.Core/Exceptions/InfluxException.cs#L108). Without that I just got a method not found exception and couldn't even see the one for the timeout.  Fun stuff...)

Thanks again for your help!
